ethereum/core/src/main/java/org/hyperledger/besu/ethereum/core/Transaction.java

```diff
@@ -639,7 +649,12 @@ public Wei getUpfrontGasCost(final Wei gasPrice) {
     if (gasPrice == null || gasPrice.isZero()) {
       return Wei.ZERO;
     }
-    return Wei.of(getGasLimit()).multiply(gasPrice);
+    var cost = BigInteger.valueOf(getGasLimit()).multiply(gasPrice.getAsBigInteger());
+    if (cost.bitLength() > 256) {
```

Comment:
how is it possible ?

Reply:
Multiplication of gas price by gas limit.  By using BigInteger I avoid the 256-bit rollover the `Wei` type enforces.  Detecting fixed-bit rollover for multiplication has too many corner cases, better to just do it in unbounded bit length and check for size.  

The `HighGasPrice` reference tests explicitly trip this condition and require the transaction to be marked as invalid.